sensor/common/image/service_impl_test.go

```diff
@@ -0,0 +1,280 @@
+			expectedError:    nil,
+			expectedResponse: createImageResponse(imageName, imageID),
+		},
+		"Cache miss and central is unreachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventOfflineMode,
+			expectCache:      expectCacheHelper(suite.mockCache, 1, nil),
+			expectedError:    errCentralNoReachable,
+			expectedResponse: nil,
+		},
+		"Cache miss and central is reachable": {
+			request:           createImageRequest(imageName, imageID, false),
+			notify:            common.SensorComponentEventCentralReachable,
+			expectCache:       expectCacheHelper(suite.mockCache, 1, nil),
+			expectRegistry:    expectRegistryHelper(suite.mockRegistryStore, 1, false),
+			expectCentralCall: expectCentralCall(suite.mockCentral, 1, createScanImageInternalResponse(imageName, imageID), nil),
+			expectedError:     nil,
+			expectedResponse:  createImageResponse(imageName, imageID),
+		},
+		"Cache miss, central is reachable and returns error": {
+			request:           createImageRequest(imageName, imageID, false),
+			notify:            common.SensorComponentEventCentralReachable,
+			expectCache:       expectCacheHelper(suite.mockCache, 1, nil),
+			expectRegistry:    expectRegistryHelper(suite.mockRegistryStore, 1, false),
+			expectCentralCall: expectCentralCall(suite.mockCentral, 1, nil, err),
+			expectedError:     errCentral,
+			expectedResponse:  nil,
+		},
+		"Cache miss, local scan, central is unreachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventOfflineMode,
+			expectCache:      expectCacheHelper(suite.mockCache, 1, nil),
+			expectedError:    errCentralNoReachable,
+			expectedResponse: nil,
+		},
+		"Cache miss, local scan, central is reachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventCentralReachable,
+			expectCache:      expectCacheHelper(suite.mockCache, 1, nil),
+			expectRegistry:   expectRegistryHelper(suite.mockRegistryStore, 1, true),
+			expectLocalScan:  expectLocalScan(suite.mockLocalScan, 1, createScannedImage(imageName, imageID), nil),
+			expectedError:    nil,
+			expectedResponse: createImageResponse(imageName, imageID),
+		},
+		"Cache miss, local scan returns error, central is reachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventCentralReachable,
+			expectCache:      expectCacheHelper(suite.mockCache, 1, nil),
+			expectRegistry:   expectRegistryHelper(suite.mockRegistryStore, 1, true),
+			expectLocalScan:  expectLocalScan(suite.mockLocalScan, 1, nil, err),
+			expectedError:    errLocalScan,
+			expectedResponse: nil,
+		},
+	}
+	for testName, c := range cases {
+		suite.T().Run(testName, func(t *testing.T) {
+			suite.service.centralReady.Reset()
```

Comment:
Could this be a source of flakiness? You're sharing the service impl amongst all table tests, perhaps it would be better to create the service instance in the beginning of the sub test to avoid having to manually reset the signal here?

Reply:
Maybe. I've changed the test to recreate the service in each subtest.